neo4j/instance_store.go

```diff
@@ -85,3 +86,54 @@ func (c *Neo4j) AddVersionDetailsToInstance(ctx context.Context, instanceID stri
 	log.InfoCtx(ctx, "neoClient AddVersionDetailsToInstance: update successful", data)
 	return nil
 }
+
+func (c *Neo4j) SetInstanceIsPublished(ctx context.Context, instanceID string) error {
+	data := log.Data{
+		"instance_id": instanceID,
+	}
+
+	log.InfoCtx(ctx, "neoClient SetInstanceIsPublished: attempting to set is_published property on instance node", data)
+
+	conn, err := c.Pool.OpenPool()
+	if err != nil {
+		return errors.WithMessage(err, "neoClient SetInstanceIsPublished: error opening neo4j connection")
+	}
+
+	defer conn.Close()
```

Comment:
Would it be safer to `defer conn.Close()` prior to checking error and possibly returning early?

Reply:
The error return value is always checked first - if an error is returned then the `conn` will be nil.

Reply:
Done 👍